### DISCUSSION TITLE
Can't install the latest version of plugins

### DISCUSSION THREAD

**Discussion by 01Kuzma** [wintercms/winter#851] [Q&A] [answered]
Hi!
I'm still trying to succeed in October upgrade to Winter.
Now the problem lies in the outdated plugin versions.

As for example here is an [amazing plugin](https://github.com/OFFLINE-GmbH/oc-responsive-images-plugin) for responsive images

On my October CMS the version `2.5.4` was of this plugin. In Winter's admin UI I pressed the button to look for the updates, but they were not found. Then I pressed the button "install plugin", and selected it from the list and now the much elder version - `2.3.11` is installed. Is it related to the marketplace? Config? How could the newest plugins be installed?

**Comment by mjauvin** [comment, marked as answer]
You need to use composer, the marketplace is not available yet. The plugins in the backend are a snapshot from when we forked OctoberCMS and are not up2date.

On the command-line, use the following to install the latest version of this plugin:
```
composer require offline/oc-responsiveimages-plugin
```

**Comment by 01Kuzma** [comment]
@mjauvin thank you for the quick reply.
I've run the command, and it seems that it had an updated plugin, but in the admin area it's still 2.3.11
The `php artisan plugin:list` also shows the 2.3.11
![image](https://user-images.githubusercontent.com/569504/222139490-ad23783a-082f-4de1-8c1b-a9275aed70cd.png)

**Comment by mjauvin** [comment]
The newest version reported on his repo is 2.4.7 but the updates/version.yaml file shows 2.5.5 as the latest.

Try this:
```
composer require offline/oc-responsiveimages-plugin@dev-develop
```

**Comment by 01Kuzma** [comment]
Doesn't help :(
On [this page](https://octobercms.com/plugin/offline-responsiveimages) the current version is displayed as `2.5.5`
![image](https://user-images.githubusercontent.com/569504/222143055-fa41de36-559d-4a0a-8857-907679022016.png)

**Comment by mjauvin** [comment]
Try this:
```
composer require offline/oc-responsiveimages-plugin dev-develop
```

**Comment by 01Kuzma** [comment]
the same 2.3.11 :(
![image](https://user-images.githubusercontent.com/569504/222147334-0c3c60c9-fffa-493c-8744-f3ff6f9e355f.png)

**Reply by LukeTowers** [reply]
Run `artisan migrate` or `artisan winter:up` after running `composer update`.

**Comment by 01Kuzma** [comment]
@LukeTowers it seems working. So, what is the correct subsequence for any possible future update?
1. Run `composer require offline/oc-responsiveimages-plugin dev-develop`
2. Then `composer update`
3. `artisan migrate` or `artisan winter:up` ?

The parameter `dev-develop` pulls the development branch from git's repo, if it exists with that name?

**Reply by WebVPF** [reply]
## Updating packages

Before installing a new plugin, it is recommended to first update the application packages:

```bash
composer update
```

## Installing plugins

Then the package (plugin) is installed:

```bash
composer require ....................
```
 
Then we run the migrations:

```bash
php artisan winter:up
```

In the future, to update the command `composer update`

**Reply by LukeTowers** [reply]
that's correct @01Kuzma, `artisan migrate` and `artisan winter:up` are just aliased versions of the same command.

**Comment by 01Kuzma** [comment]
One more additional question: as I'm running a shared hosting, I'm unable to run `composer` directly on the hosting.
So, I'm running `composer:update + artisan migrate` locally, then reupload files to hosting.
In order to not copy everything round and round which folders could be skipped? 
Only `storage`  folder?

**Reply by bennothommo** [reply]
You can pretty much just copy just the `modules`, `plugins`, and `themes` directories. Those are the only folders that'll be directly affected by Composer updates.

**Reply by LukeTowers** [reply]
and the `vendor` folder @01Kuzma

**Reply by bennothommo** [reply]
Whoops, @LukeTowers is correct - `vendor` too :)
